### PR TITLE
Fixed net_graph() unused wires problem (issue #204) and added a test for it

### PR DESCRIPTION
**Pull request by nvandervoort**

### CODE REVIEW

**Comment by Gamrix** [comment]
Umm, why is the probe commit also in this branch???

File: pyrtl/inputoutput.py [hunk 1]
```diff
@@ -251,10 +251,12 @@ def net_graph(block=None, split_state=False):
...
     graph = {}
+    valid_wires = set()
 
     # add all of the nodes
     for net in block.logic:
         graph[net] = {}
+        valid_wires.update(net.args, net.dests)
 
     wire_src_dict, wire_dst_dict = block.net_connections()
     dest_set = set(wire_src_dict.keys())
```
Comment (Gamrix) [inline pyrtl/inputoutput.py]: simple question: what is `dest_set`?

**Comment by nvandervoort** [comment]
Had to cherry-pick it dawg

File: pyrtl/inputoutput.py (hunk 1 above)
Comment (nvandervoort) [inline pyrtl/inputoutput.py]: The set of all destination wires in the current block?

Comment (Gamrix) [inline pyrtl/inputoutput.py]: that's right. Can you think of a way to simplify your code?

**Comment by Gamrix** [comment]
How do you prevent things like the probe commit from being in the branch in the future?

**Comment by nvandervoort** [comment]
I dunno

**Pull request closed without merging**